lib/cylc/task_proxy.py

```diff
@@ -623,33 +732,110 @@ def handle_event(
         if self.tdef.run_mode != 'live':
             return
 
-        handlers = self.event_hooks[event + ' handler']
-        if handlers:
-            self.log(DEBUG, "Queueing " + event + " event handler(s)")
-            for handler in handlers:
-                self.log(DEBUG, "Queueing " + event + " event handler")
-                cmd = ""
-                env = None
-                if TaskProxy.event_handler_env:
-                    env = dict(os.environ)
-                    env.update(TaskProxy.event_handler_env)
+        self.retrieve_job_logs(event, message)
+        self.send_event_mail(event, message)
+        self.call_event_handlers(event, message)
+
+    def retrieve_job_logs(self, event, _=None):
+        """Retrieve remote job logs."""
+        key = (self.JOB_LOGS_RETRIEVE, event, "%02d" % self.submit_num)
+        if (key in self.event_handler_try_states or
+                event not in ["failed", "retry", "succeeded"]):
+            return
+        if self.user_at_host in [user + '@localhost', 'localhost']:
+            self.register_job_logs(self.submit_num)
+            return
+        conf = self.tdef.rtconfig["remote"]
+        size = conf["retrieve job log max size"]
+        if not size:
+            return
```

Comment:
This isn't very intuitive (I was wondering why the new handler wasn't being called!).  Maybe set a sensible default size.